### PR TITLE
CategorySearchService does not respect ResponseGroup

### PR DESCRIPTION
**Pull request by AliveMen**
reduce search result
add ReduceDetails method

### CODE REVIEW

File: VirtoCommerce.CatalogModule.Web.Core/Model/Category.cs [hunk 1]
```diff
@@ -35,13 +35,14 @@ public class Category : AuditableEntity, ISeoSupport, IHasOutlines
         /// The code.
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web.Core/Model/Category.cs]: Are there significant changes? if no, need to revert.

File: VirtoCommerce.CatalogModule.Data/Search/ProductSearchService.cs [hunk 2]
```diff
@@ -45,71 +45,30 @@ protected virtual ItemResponseGroup GetResponseGroup(ProductSearchCriteria crite
...
-                {
-                    product.Assets = null;
-                }
-
-                if (!responseGroup.HasFlag(ItemResponseGroup.ItemProperties))
-                {
-                    product.Properties = null;
-                }
-                else if(!string.IsNullOrEmpty(criteria.LanguageCode))
+
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Data/Search/ProductSearchService.cs]: Shouldn't we call product.ReduceDetails here and in CategorySearchService?

Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Data/Search/ProductSearchService.cs]: In this place product type is Web.Models.Product and haven't got ReduceDetails method. By using toWebModel method we reducing details.

File: VirtoCommerce.CatalogModule.Web.Core/Model/Category.cs (hunk 1 above)
Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Web.Core/Model/Category.cs]: reverted

File: VirtoCommerce.CatalogModule.Data/Search/CategorySearchService.cs [hunk 3]
```diff
@@ -38,6 +38,7 @@ protected override IList<Category> LoadMissingItems(string[] missingItemIds, Cat
 
         protected override void ReduceSearchResults(IEnumerable<Category> items, CategorySearchCriteria criteria)
         {
+
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Search/CategorySearchService.cs]: Why???

File: VirtoCommerce.CatalogModule.Data/Search/ProductSearchService.cs (hunk 2 above)
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Search/ProductSearchService.cs]: Remove empty line

File: VirtoCommerce.CatalogModule.Data/Services/ItemServiceImpl.cs [hunk 4]
```diff
@@ -140,47 +140,8 @@ protected virtual void ReduceDetails(CatalogProduct product, ItemResponseGroup r
...
-            if (!respGroup.HasFlag(ItemResponseGroup.Seo))
-            {
-                product.SeoInfos = null;
-            }
-            if (!respGroup.HasFlag(ItemResponseGroup.Variations))
-            {
-                product.Variations = null;
-            }
+            product.ReduceDetails(respGroup.ToString());
+
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Services/ItemServiceImpl.cs]: Remove

File: VirtoCommerce.CatalogModule.Web.Core/Model/Product.cs [hunk 5]
```diff
@@ -329,5 +329,66 @@ public virtual CatalogProduct ToModel(CatalogProduct product)
...
+            }
+
+            if (Variations != null)
+            {
+                //For nested variations leave only variation properties to decrease resulting JSON
+                foreach (var variation in Variations)
+                {
+                    if (variation.Properties != null)
+                    {
+                        variation.Properties =
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Web.Core/Model/Product.cs]: Don't split this line here. Move `.Where` and `.ToList` to different lines.

File: VirtoCommerce.CatalogModule.Web.Core/Model/Product.cs [hunk 6]
```diff
@@ -329,5 +329,67 @@ public virtual CatalogProduct ToModel(CatalogProduct product)
...
+            var productResponseGroup = EnumUtility.SafeParseFlags(responseGroup, ItemResponseGroup.ItemLarge);
+
+            if (!productResponseGroup.HasFlag(ItemResponseGroup.ItemAssets))
+            {
+                Assets = null;
+            }
+
+            if (!productResponseGroup.HasFlag(ItemResponseGroup.ItemProperties))
+            {
+                Properties = null;
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web.Core/Model/Product.cs]: Clear PropertyValues.
In general it should be like before for model: https://github.com/VirtoCommerce/vc-module-catalog/pull/226/files#diff-a739380ca84d772b1756c55fdcbec922L143

File: VirtoCommerce.CatalogModule.Web.Core/Model/Product.cs [hunk 7]
```diff
@@ -329,5 +329,67 @@ public virtual CatalogProduct ToModel(CatalogProduct product)
...
+            {
+                SeoInfos = null;
+            }
+
+            if (!productResponseGroup.HasFlag(ItemResponseGroup.Variations))
+            {
+                Variations = null;
+            }
+
+            if (Variations != null)
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web.Core/Model/Product.cs]: No need in this section.

File: VirtoCommerce.CatalogModule.Web.Core/Model/Product.cs [hunk 8]
```diff
@@ -329,5 +329,67 @@ public virtual CatalogProduct ToModel(CatalogProduct product)
...
+                        variation.Properties = variation.Properties
+                            .Where(x => x.Type == PropertyType.Variation)
+                            .ToList();
+                    }
+
+                    variation.Outlines = null;
+                    variation.Reviews = null;
+                }
+            }
+        }
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web.Core/Model/Product.cs]: Need to clear ReferencedAssociations and Inventory

**Pull request merged**